src/vec_map.rs

```diff
@@ -0,0 +1,50 @@
+use serde::{ser::SerializeMap, Serialize, Serializer};
+use std::collections::HashMap;
+
+/// Use a sorted Vec to store key-value pairs.
+/// Useful as a replacement of HashMap/BTreeMap where:
+/// - There is no need for fast access of values by keys.
+/// - The keys need to be sorted to be use as partition key
+/// for PartitionBatchSink.
+///
+/// Ideal to store labels map in various sinks.
```

Comment:
I want to try this simple data structure, also useful in loki sink (https://github.com/timberio/vector/pull/5973#discussion_r563002293).